eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -490,4 +499,12 @@ object Peer {
       walletStaticPaymentBasepoint = walletStaticPaymentBasepoint,
       initFeatures = initFeatures)
   }
+
+  sealed trait RelayPolicy
+
+  case object NoRelay extends RelayPolicy
+
+  case object RelayChannelsOnly extends RelayPolicy
+
+  case object RelayAll extends RelayPolicy
```

Comment:
I think this would be better inside `MessageRelay` (or rename it `OnionMessageRelayPolicy` if you'd rather keep it here)?
I guess you put it in `Peer` to potentially later reuse them for gossip policies, but I think it will be confusing as gossip won't have exactly the same behavior as onion messages, so it's probably better to introduce a slightly different set of relay policies for gossip later.

Reply:
Done